### PR TITLE
Typescript migration

### LINKED ISSUE #152: Migrate to typescript

### PR DESCRIPTION
**Pull request by Brainicism**
Closes #152

### CODE REVIEW

File: src/helpers/discord_utils.ts [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        fs.closeSync(fs.openSync(filePath, "w"));
+    }
+}
+
+const arraysEqual = (arr1: Array<any>, arr2: Array<any>): boolean => {
+    if (arr1.length !== arr2.length) {
+        return false;
+    }
+
+    var arr1 = arr1.concat().sort();
```
Comment (taahamahdi) [inline src/helpers/discord_utils.ts:132]: Any reason why these can't be `let`s? Sorry, I know this was from something I had approved earlier, sorry about that.

File: package.json [hunk 2]
```diff
@@ -4,7 +4,8 @@
   "main": "src/kmq.js",
   "scripts": {
     "seed": "cd seed && node seed_db.js",
-    "start": "cd src/ && node kmq.js"
+    "dev": "cd src && ts-node kmq",
+    "prod": "tsc && cp -r ./config ./build && cd build/src && node kmq.js"
```
[the remark this answers is not in the document]
Comment (Brainicism) [reply package.json:8]: Fixed

File: src/kmq.ts [hunk 3]
```diff
@@ -86,15 +89,15 @@ client.on("voiceStateUpdate", (oldState, newState) => {
             }
         }
         // Bot was disconnected by another user
-        if (!oldUserChannel.members.has(client)) {
+        if (!oldUserChannel.members.has(client.user.id)) {
```
[the remark this answers is not in the document]
Comment (Brainicism) [reply src/kmq.ts:92]: Still works fine

File: src/helpers/discord_utils.ts (hunk 1 above)
Comment (Brainicism) [reply src/helpers/discord_utils.ts:132]: copied this from stackoverflow, fixed

File: src/kmq.ts [hunk 4]
```diff
@@ -104,8 +107,10 @@ const getGuildPreference = (guildPreferences, guildID) => {
     return guildPreferences[guildID];
 }
 
-const parseMessage = (message, botPrefix) => {
-    if (message.charAt(0) !== botPrefix) return null;
+
+
+const parseMessage = (message: string, botPrefix: string): ParsedMessage => {
+    // if (message.charAt(0) !== botPrefix) return null;
```
[the remark this answers is not in the document]
Comment (Brainicism) [reply src/kmq.ts:113]: Good catch

**Comment by taahamahdi** [comment]
All is good. Ready to merge when you want.

**Pull request merged**